segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -1602,6 +1641,76 @@ private void read00(RevisionDataInput input, TruncationRecord.TruncationRecordBu
         }
     }
 
+    /**
+     * Journal record for segment append.
```

Comment:
I don't understand how this ties into what is described in the PR. If we're just removing an optimization, which was only sometimes on, why is there a new journal record type?

Reply:
see below...